### PR TITLE
Jak & Daxter Client : queue game text messages to get items faster during release

### PR DESCRIPTION
**Pull request by t-rbernard**
## What is this fixing or adding?
Fix items taking a long time to start being received when a world is released due to many json messages needing to be processed before the ItemReceived message is available in the message queue. 

To do this, when json messages go through the J&D Client class, the messages are put in a queue that's processed during the repl main_tick. This way, we limit the time taken by repl client sending messages to the game and get to the ItemReceived message much faster

## How was this tested?
Multiple generations and release of J&D games with global orbsanity (~2000 items released)

## If this makes graphical changes, please attach screenshots.
N/A

### CODE REVIEW

**Comment by massimilianodelliubaldini** [comment]
```
1 Orbsanity, Time from "/release Markus" to the ingame orb counter reading 2000

              Total     Time to Receiving Items

    Markus: 10m 49s     4m 26s

    Troyoo:  9m 18s        25s
```

**Comment by massimilianodelliubaldini** [comment]
I'm only holding off on merging this until I can verify a fix for the v0.1.0 debacle.

**Comment by t-rbernard** [comment]
Thank you for the review and added stats ! I didn't expect the full time to receiving all orbs to be almost as long, but I guess all the little holdups to treat the message queue do accumulate over time and make it so it takes almost as long in the end :thinking: 

The little holdup for a single message was expected yeah since we wait for the main_tick. 
A more in-depth fix that would also prevent any message holdup would probably have to involve a big refactor to how messages are handled to tie in-game messaging to the items being received instead of forwarding json_messages, but we'd probably lose all info on who sent the item (which is pretty neat to have). 
That'd sync in-game messaging to receiving items as well as (probably) being even faster than this, but it's way more involved and probably unnecessary considering the results shown here

**Review by massimilianodelliubaldini — CHANGES_REQUESTED** [review]

**Comment by massimilianodelliubaldini** [comment]
Still getting used to GitHub's review process, forgive the mess.

File: worlds/jakanddaxter/client/ReplClient.py
Comment (t-rbernard) [reply worlds/jakanddaxter/client/ReplClient.py:48]: yeah it shouldn't cause any issue. I kept it inside the ReplClient class because I didn't see it being used outside of the ReplClient class, so it didn't need to be visible outside of it, but if you prefer it outside for style/clarity it shouldn't change much

**Comment by t-rbernard** [comment]
no worries, I'm quite new to it to, so forgive the commit spam for minor style changes 😅

File: worlds/jakanddaxter/client/ReplClient.py
Comment (massimilianodelliubaldini) [reply worlds/jakanddaxter/client/ReplClient.py:48]: It's entirely a personal thing - in general I try to avoid nested class definition, even if they make perfect sense in a language like Python that doesn't have access modifiers (public/private/etc). But maybe that's just crazy talk and there's no reason for me to work like that. I'll leave it up to you if you want to change it or not, just let me know either way.

Comment (t-rbernard) [reply worlds/jakanddaxter/client/ReplClient.py:48]: Would you rather have it in the same file but not in the ReplClient class or in its own little file to import in ReplClient ?

Comment (massimilianodelliubaldini) [reply worlds/jakanddaxter/client/ReplClient.py:48]: Same file separate classes is fine by me. MemoryReader.py has this setup as well.

Comment (t-rbernard) [reply worlds/jakanddaxter/client/ReplClient.py]: Aight, it does look a bit cleaner compared to having it kinda in the middle of the the ReplClient class member initialization so I moved it just outside ReplClient

**Review by massimilianodelliubaldini — APPROVED** [review]

**Pull request merged**
